### ISSUE TITLE
Google Suite: retrieving user's profile picture?

### ISSUE THREAD

**Issue by kahuitel** [catalyst/moodle-auth_saml2#333]
I've successfully setup moodle-auth_saml2 with Moodle 3.7, email, firstname, lastname and a custom field is being fetched OK, but I cannot figure out how to get user's URL.

Is it something possible with SAML2 ?

Thank you,

FX

**Comment by theopen-institute** [comment]
Did you ever find a solution to this? I am also very interested in updating user profile pictures.

**Comment by brendanheywood** [comment]
Note that the mapping of user attributes from any auth plugin (saml or db, cas etc) is all handled by moodle core, and moodle core can only set the user profile fields which doesn't include the users picture url. This seems like a great addition though, and would best be done as a tracker in moodle itself. If anyone wants to sponsor this feature we could sort that, please contact us and quote this issue:

https://www.catalyst-au.net/contact-us

**Comment by theopen-institute** [comment]
Thanks for the clarification @brendanheywood. I’ll reach out to Catalyst

**Comment by jtsafran** [comment]
Did this ever get a related tracker?  The core oAuth plugin manages to add it, so it would seem Moodle could configure that as part of the core auth API.  I did a quick search and I am not seeing it.  Would like to upvote it, if it exists!!

**Comment by brendanheywood** [comment]
Yeah it's annoying they've gone down the path of directly implementing this in oauth2 rather than making a new api that any auth plugin can use. saml2 could do a cut and paste job but its better if we could refactor that into a new api.

https://github.com/moodle/moodle/blob/master/auth/oauth2/classes/auth.php#L229-L292

I've made a tracker here:

https://tracker.moodle.org/browse/MDL-74219
